### PR TITLE
Feat: Add friend on left menu

### PR DESCRIPTION
**Pull request by starbrdi**
https://github.com/thenewboston-developers/Account-Manager/issues/111

### CODE REVIEW

**Comment by angle943** [comment]
@starbrdi great work. One thing, let's use `accountNumber` instead of `friendNumber`

File: src/renderer/utils/store.ts [hunk 1]
```diff
@@ -135,12 +145,19 @@ export function unsetDataReducer() {
...
   };
 }
 
+export function unsetFriendLocalAndStateReducer() {
+  return (state: any, {payload: {friend_number: friendNumber}}: PayloadAction<ManagedFriend>) => {
+    delete state[friendNumber];
+    localStore.set(getStateName(MANAGED_FRIENDS), state);
+  };
+}
+
```
Comment (angle943) [inline src/renderer/utils/store.ts:160]: once friendNumber becomes accountNumber, this function is no longer needed

File: src/renderer/utils/store.ts [hunk 2]
```diff
@@ -38,6 +39,15 @@ export function setAccountLocalAndStateReducer<T extends AccountNumber>() {
...
 }
 
+export function setFriendLocalAndStateReducer<T extends FriendNumber>() {
+  return (state: any, {payload}: PayloadAction<T>) => {
+    const {friend_number: friendNumber} = payload;
+    const account = state[friendNumber];
+    state[friendNumber] = account ? {account, ...payload} : payload;
+    localStore.set(getStateName(MANAGED_FRIENDS), state);
+  };
+}
```
Comment (angle943) [inline src/renderer/utils/store.ts:49]: once friendNumber becomes accountNumber, this function is no longer needed

File: src/renderer/types/network.ts [hunk 3]
```diff
@@ -71,6 +71,10 @@ export interface Error {
   error: string | null;
 }
 
+export interface FriendNumber {
+  friend_number: string;
+}
```
Comment (angle943) [inline src/renderer/types/network.ts:76]: every occurence of this type can be replace with `AccountNumber`

File: src/renderer/containers/Layout/index.tsx [hunk 4]
```diff
@@ -24,6 +24,9 @@ export const Layout: FC = () => {
           <Route path="/account/:accountNumber">
             <Account />
           </Route>
+          <Route path="/friend/:friendNumber">
```
Comment (angle943) [inline src/renderer/containers/Layout/index.tsx:27]: :accountNumber

File: src/renderer/containers/Friend/index.tsx [hunk 5]
```diff
@@ -1,11 +1,80 @@
...
+import {getManagedFriends} from '@renderer/selectors';
 import './Friend.scss';
 
 const Friend: FC = () => {
+  const {friendNumber} = useParams();
+  const {path, url} = useRouteMatch();
+  const [deleteModalIsOpen, toggleDeleteModal] = useBooleanState(false);
+  const [sendPointsModalIsOpen, toggleSendPointsModal] = useBooleanState(false);
+  const managedFriends = useSelector(getManagedFriends);
+  const managedFriend = managedFriends[friendNumber];
```
Comment (angle943) [inline src/renderer/containers/Friend/index.tsx:22]: unused variable

File: src/renderer/containers/Friend/FriendOverview/index.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  useEffect(() => {
+    if (!activePrimaryValidator) return;
+
+    const fetchData = async (): Promise<void> => {
+      const {ip_address: ipAddress, port, protocol} = activePrimaryValidator;
+      const address = formatAddress(ipAddress, port, protocol);
+
+      setLoading(true);
+      const {data} = await axios.get(`${address}/friend/${friendNumber}/balance`);
```
Comment (angle943) [inline src/renderer/containers/Friend/FriendOverview/index.tsx:29]: should be `${address}/accounts/${accountNumber}/balance` per our discussion on slack

File: src/renderer/utils/store.ts (hunk 1 above)
Comment (starbrdi) [reply src/renderer/utils/store.ts]: @angle943 This function is necessary, isn't it?

**Pull request merged**